### PR TITLE
Replace file editor with openhands-aci

### PR DESCRIPTION
**Pull request by ryanhoangt**
**End-user friendly description of the problem this fixes or functionality that this introduces**

- [ ] Include this change in the Release Notes. If checked, you must provide an **end-user friendly** description for your change below

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

This PR is to:
- [x] Verify feature parity of [`openhands-aci`](https://github.com/All-Hands-AI/openhands-aci) with the existing file editor
- [x] Integrate `openhands-aci` into OpenHands

---
**Link of any specific issues this addresses**

### CODE REVIEW

**Comment by ryanhoangt** [comment]
@xingyaoww I have tried to test via the UI and it looks good to me. Your existing unit tests also passed, with minor changes in error messages. I think it's ready for eval now

**Comment by ryanhoangt** [comment]
(1) We got the same success rate with CodeAct v2.2:

Screenshot 2024-11-10 at 12 38 40

File: pyproject.toml [hunk 1]
```diff
@@ -61,6 +61,7 @@ protobuf = "^4.21.6,<5.0.0" # chromadb currently fails on 5.0+
 opentelemetry-api = "1.25.0"
 opentelemetry-exporter-otlp-proto-grpc = "1.25.0"
 modal = "^0.64.145"
+openhands-aci = {git = "https://github.com/All-Hands-AI/openhands-aci.git", rev = "ht/simplify-undo-edit"}
```
Comment (xingyaoww) [inline pyproject.toml:64]: Can you cut a release and set this as openhands-aci 0.1.0?

File: tests/unit/test_agent_skill.py [hunk 2]
```diff
@@ -781,7 +781,7 @@ def test_file_editor_create(tmp_path):
     assert result is not None
```
Comment (xingyaoww) [inline tests/unit/test_agent_skill.py:781]: I'm thinking if we can start moving these unit tests out of OpenHands repo and into openhands-aci 🤔

**Review by xingyaoww — COMMENTED** [review]
Left a few nit - otherwise LGTM! This is a awesome start to cleanup our codebase :)

File: tests/unit/test_agent_skill.py (hunk 2 above)
Comment (ryanhoangt) [reply tests/unit/test_agent_skill.py:780]: Done!

**Review by xingyaoww — APPROVED** [review]

**Pull request merged**